doc/specs/stdlib_sorting.md

```diff
@@ -0,0 +1,644 @@
+
+    * Partitions the remaining array using a median of three,
+    * Calls `introsort` proper on the leftmost partition,
+    * Calls `introsort` proper on the rightmost partition, and then
+      returns.
+
+The resulting algorithm is of order O(N Ln(N)) run time performance
+for all inputs. Because it relies on `quicksort`, the coefficient of
+the O(N Ln(N)) behavior is typically small compared to other sorting
+algorithms on random data. On partially sorted data it can show either
+slower `heap sort` performance, or enhanced performance by up to a
+factor of six. Still, even when it shows enhanced performance, its
+performance on partially sorted data is typically an order of
+magnitude slower than `ORD_SORT`. Its memory requirements are also
+low, being of order O(Ln(N)), while the memory requirements of
+`ORD_SORT` and `SORT_INDEX` are of order O(N).
+
+### Tentative specifications of the `stdlib_sorting` procedures
+
+#### `ord_sort` - sorts an input array
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns an input `array` with the elements sorted in order of
+increasing value.
+
+##### Syntax
+
+`call [[stdlib_sorting(module):ord_sort(subroutine)]]ord_sort ( array[, work ] )`
+
+##### Class
+
+Generic subroutine.
+
+##### Arguments
+
+`array` : shall be a rank one array of any of the types:
+`integer(int8)`, `integer(int16)`, `integer(int32)`, `integer(int64)`,
+`real(sp)`, `real(dp)`, `real(qp)`, `character(*)`, or
+`type(string_type)`. It is an `intent(inout)` argument. On input it is
+the array to be sorted. If both the type of `array` is real and at
+least one of the elements is a `NaN`, then the ordering of the result
+is undefined. Otherwise on return its elements will be sorted in order
+of non-decreasing value.
+
+`work` (optional): shall be a rank one array of the same type as
+array, and shall have at least `size(array)/2` elements. It is an
+`intent(inout)` argument. It is intended to be used as "scratch"
+memory for internal record keeping. If associated with an array in
+static storage, its use can significantly reduce the stack memory
+requirements for the code. Its contents on return are undefined.
+
+##### Notes
+
+`ORD_SORT` implements a hybrid sorting algorithm combining
+`merge sort`, and `insertion sort`. For most purposes it behaves like
+a `merge sort`, providing worst case `O(N Ln(N))` run time performance
+for most random arrays, that is typically slower than `SORT`.
+However, if the array has significant runs of decreasing or
+non-decreasing values, performance can be much better than
+`SORT`, with `O(N)` behavior on uniformly decreasing, or
+non-decreasing arrays. The optional `work` array replaces "scratch"
+memory that would otherwise be allocated on the stack. If `array` is of
+any type `REAL` the order of its elements on return undefined if any
+element of `array` is a `NaN`.  Sorting of `CHARACTER(*)` and
+`STRING_TYPE` arrays are based on the operator `>`, and not on the
+function `LGT`.
```

Comment:
Should links to `>` and `lgt` be provided?

Reply:
Maybe. How do I do it?